### ISSUE TITLE
Weight exploding when running  Swivel with 8-GPUs

### ISSUE THREAD

**Issue by sweaterr** [tensorflow/models#1170]
Hello
I'm running the Swivel algorithm on my data. However, in single gpu, learning can be done without any problem, but when swivel is executed with 8-gpus, loss and weight become larger at about 100k global steps as follows, and learning can not be done properly.

![image](https://cloud.githubusercontent.com/assets/1518919/23933535/80ed00d6-0982-11e7-9b2b-149ffb4fb525.png)

environments
* data size: 120gb
* os: centos 7
* tensorflow: 1.0.1
* gpu:m40*8

parameters
```
num_gpus=8
num_concurrent_steps=8
learning_rate=1.0
vocab_size: 5m
shard_size: 10000
embedding_size=128
num_submatrices : 250000
```
What can I do for this?
Thanks in advance.

**Comment by concretevitamin** [comment]
Can you clarify what exactly is the code you're running?  It'd be great if you can also paste the cmdline you used.

**Comment by sweaterr** [comment]
I modified the code in  [`tensorflow/models/swivel/swivel.py`](https://github.com/tensorflow/models/blob/master/swivel/swivel.py) as follows.
```python
flags.DEFINE_integer('embedding_size', 128, 'Size of the embeddings')
flags.DEFINE_boolean('trainable_bias', False, 'Biases are trainable')
flags.DEFINE_integer('submatrix_rows', 10000, 'Rows in each training submatrix. '
                     'This must match the training data.')
flags.DEFINE_integer('submatrix_cols', 10000, 'Rows in each training submatrix. '
                     'This must match the training data.')
flags.DEFINE_float('loss_multiplier', 1.0 / 10000,
                   'constant multiplier on loss.')
flags.DEFINE_float('confidence_exponent', 0.5,
                   'Exponent for l2 confidence function')
flags.DEFINE_float('confidence_scale', 0.25, 'Scale for l2 confidence function')
flags.DEFINE_float('confidence_base', 0.1, 'Base for l2 confidence function')
flags.DEFINE_float('learning_rate', 1.0, 'Initial learning rate')
flags.DEFINE_integer('num_concurrent_steps', 8,
                     'Number of threads to train with')
flags.DEFINE_integer('num_readers', 100,
                     'Number of threads to read the input data and feed it')
flags.DEFINE_float('num_epochs', 10, 'Number epochs to train for')
flags.DEFINE_float('per_process_gpu_memory_fraction', 0,
                   'Fraction of GPU memory to use, 0 means allow_growth')
flags.DEFINE_integer('num_gpus', 8,
                     'Number of GPUs to use, 0 means all available')

```
and run as follows
```
./swivel.py --input_base_path $my_input_path   --output_base_path data/swivel_data
```

**Comment by concretevitamin** [comment]
@martinwicke: I can't seem to find a clear owner for models/swivel.  Since you added the initial Github commit, could you recommend someone to answer this?

**Comment by concretevitamin** [comment]
Actually, @sweaterr I think this issues belong to the model-specific mailing list, which most likely will be monitored by the owners.  See: https://github.com/tensorflow/models/tree/master/swivel#contact.
